src/main/java/com/minecolonies/structures/helpers/Settings.java

```diff
@@ -202,4 +205,34 @@ public BlockPos getOffset(final PlacementSettings settings)
         }
         return new BlockPos(0, 0, 0);
     }
+
+    /**
+     * Makes the building mirror.
+     */
+    public void mirror()
+    {
+        if (structure == null)
+        {
+            return;
+        }
+        isMirroed = !isMirroed;
+
+        structure.setPlacementSettings(structure.getSettings().setMirror(getMirror()));
```

Comment:
I don't think we should be doing this. Let's handle mirroring the same way as rotation